### PR TITLE
Entirety of the Maflipulator Subsystem

### PR DESCRIPTION
**Pull request by DavidDesfosse**
Has the maflipulator which differentiates between sides, restricts the movement so that it doesn't hit the lift, has PID, and is able to flip from one side of the robot to the other. Some minor changes to manipulator about the piston?

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Maflipulator.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            @Override
+            protected void end() {
+                flipMotor.set(0);
+                // Called once after isFinished returns true
+            }
+        };
+    }
+
+    public Command moveToPosition() {
+        return new Command("move to position", this) {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Maflipulator.java:149]: Now that we have a working `Command`, refer to [the PIDCommand documentation](http://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/command/PIDCommand.html) and see if you can convert this to use `PIDCommand` instead.

File: src/main/java/frc/robot/maps/PracticeBot.java [hunk 2]
```diff
@@ -122,14 +123,29 @@ public Encoder getLeftEncoder() {
...
+            @Override
+            public SendableSpeedController getFlipMotor() {
+                return new MockSpeedController();
             }
 
             @Override
-            public DoubleSolenoid getClimbPiston() {
-                return new DoubleSolenoid(4, 5);
+            public Potentiometer getMaflipulatorPot() {
+                return new AnalogPotentiometer(8);
```
Comment (msoucy) [inline src/main/java/frc/robot/maps/PracticeBot.java:148]: `return new MockPotentiometer();`, and fix the imports afterward

**Review by msoucy — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Maflipulator.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+            @Override
+            protected void initialize() {
+                anglePID.reset();
+                anglePID.setSetpoint(0);
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Maflipulator.java:154]: Don't we need to be able to set the setpoint to a value, not just hardcode it to 0?

File: src/main/java/frc/robot/subsystems/Maflipulator.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+    private MaflipulatorSide currentPosition;
+
+    private SendableSpeedController flipMotor;
+    private Potentiometer anglePot;
+
+    double angleCorrection;
+    PIDController anglePID;
+
+    public Maflipulator(final RobotMap.MaflipulatorMap map) { // NOPMD
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Maflipulator.java:34]: We need to figure out which side of the `lift` the `maflipulator` is on.

**Review by bot190 — CHANGES_REQUESTED** [review]

**Comment by msoucy** [comment]
Closing this in favor of #12, which has been more recently updated.

David, collaborate with Quinlan to make changes as specified in #12 and push to his repository

**Pull request closed without merging**